pow-migration/src/lib.rs

```diff
@@ -302,7 +305,24 @@ pub async fn migrate(
             )
             .await?;
 
-            log::info!("PoS Genesis generation is completed");
+            // Sort vectors for a consistent hash digest
+            genesis_config.validators.sort();
+            genesis_config.stakers.sort();
+            genesis_config.basic_accounts.sort();
+            genesis_config.vesting_accounts.sort();
+            genesis_config.htlc_accounts.sort();
+
+            let mut hasher = Blake2bHasher::new();
+            genesis_config
+                .serialize_to_writer(&mut hasher)
+                .unwrap_or_else(|error| {
+                    exit_with_error(error, "Failed to serialize genesis config")
+                });
+            genesis_block_hash = hasher.finish();
```

Comment:
Is that actually the hash of the genesis block as it will be written to the database? Or just the hash of the genesis _config_?

Reply:
Hmm.. that's then not useful after the migration anymore. Would be nice to have the actual block hash, to actually have a connection between the two chains.

Reply:
It's not designed to be used after the migration. But rather an extra filter for honest migrators to prevent that ready transactions that get send, by other validators, just before the activation window closes, accidentally fall into the next window.

This hash is by no means trustworthy in the sense that it doesn't actually have to mean that you've migrated. You can send this transaction regardless and simply mimic the hash that other validators are broadcasting.

Reply:
I get that for the purpose it serves (syncing between validators) this fits the bill perfectly. 

Just, from a bigger picture, for me it would have sentimental and "proof" value if we could write the genesis block of the PoS chain into the end of the PoW chain. It _connects_ the two and makes it more of a continuation.

Reply:
Unfortunately we cannot do it the other way round and write the transition block hash into the PoS chain, as the extra data field of the genesis block is already used to encode the genesis supply amount.

Reply:
Ohhh! Actually, I just found that the genesis block's `parentHash` is the PoW chain's transition block! Cool! :sunglasses: